### PR TITLE
Add DDRace Score NETMSG

### PR DESCRIPTION
**Pull request by fokkonaut**

### CODE REVIEW

**Comment by fokkonaut** [comment]
the server admin can edit sv_ddrace_score in his autoexec file. if its 1, it allows the client to use ddrace scoreboard, if its 0 it disallows it. this would work nicely with IsDDNet by timakro, to have a server with a custom gametype, ddrace features, but also the normal score (not ddrace score with times).
it can also be used to run mutliple gametypes on a single server without having weird time scores (need more editing serverside ofc).

with this addition we could also merge IsBlockWorlds and IsDDNet, then exec just have to add this message to tell the client that it has to use the normal score and not ddrace score.

here is an example of how we would use this feature:
https://github.com/DDNetPP/DDNetPP/pull/188

**Review by heinrich5991 — CHANGES_REQUESTED** [review]
Thank you for your updated pull request!

I think we should rename the feature a little, "DDRace score" doesn't quite capture what it does. The config variable seems a bit superfluous; it should be good enough if people can mod it (for blockworlds, DDRacePP, etc.).

If you need help with updating this pull request, ping me on Discord. :)

File: src/game/client/components/scoreboard.cpp [hunk 1]
```diff
@@ -404,7 +404,7 @@ void CScoreboard::RenderScoreboard(float x, float y, float w, int Team, const ch
 		}
 
 		// score
-		if(m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard)
+		if(m_IsGameTypeRace && g_Config.m_ClDDRaceScoreBoard && ((m_pClient->m_AllowDDRaceScore[0] && !g_Config.m_ClDummy) || (m_pClient->m_AllowDDRaceScore[1] && g_Config.m_ClDummy)))
```
[the remark this answers is not in the document]
Comment (fokkonaut) [reply src/game/client/components/scoreboard.cpp]: i dont understand this one? there need to be 2 Independent ones, one for the dummy, one for the main tee. Just like this.

[the remark this answers is not in the document]
Comment (fokkonaut) [reply src/game/client/components/scoreboard.cpp]: i think i got it, thanks :)

File: src/engine/shared/protocol_ex_msgs.h [hunk 2]
```diff
@@ -25,3 +25,4 @@ UUID(NETMSG_IDONTKNOW,    "i-dont-know@ddnet.tw")
 
 UUID(NETMSG_RCONTYPE,     "rcon-type@ddnet.tw")
 UUID(NETMSG_MAP_DETAILS,  "map-details@ddnet.tw")
+UUID(NETMSG_DDRACE_SCORE, "ddrace-score@ddnet.tw")
```
[the remark this answers is not in the document]
Comment (fokkonaut) [reply src/engine/shared/protocol_ex_msgs.h]: is `TIME_SCORE` ok?

[the remark this answers is not in the document]
Comment (heinrich5991) [reply src/engine/shared/protocol_ex_msgs.h]: Yes. :)

**Review by heinrich5991 — APPROVED** [review]

**Comment by heinrich5991** [comment]
Looks good to me. Sorry for the long waiting times.

bors r+

**Comment by fokkonaut** [comment]
i added another commit because now blockworlds can add this netmsg and disallow time score using that.

**Comment by fokkonaut** [comment]
can someone remove the last commit and merge it then, so from the state where heinrich wanted to merge it?

**Comment by def-** [comment]
Done.

That's why you should use branches when doing a PR ;)

**Comment by def-** [comment]
Done.

That's why you should use branches when doing a PR ;)

**Comment by fokkonaut** [comment]
thanks :)

**Pull request closed without merging**